model/task_lifecycle.go

```diff
@@ -865,7 +865,7 @@ func getBuildStatus(buildTasks []task.Task) string {
 	// not started
 	noStartedTasks := true
 	for _, t := range buildTasks {
-		if !evergreen.IsUnstartedTaskStatus(t.Status) {
+		if !evergreen.IsUnstartedTaskStatus(t.Status) && !t.Blocked() {
```

Comment:
So I've been trying to parse this logic for a while and it's pretty confusing to think about, so I  think we need to change the variable names (and also write a unit test to confirm that this change fixes the case described in the ticket). 

Originally we set "noStartedTasks" to false only if ANY task wasn't in the "undispatched" state (i.e. any task starts or fails). Now we're saying "if any task isn't in undispatched AND it's not blocked", but wouldn't a task that's started also be unblocked? So I'm not sure this is the right change.

Reply:
updated